### PR TITLE
Allow RSA key for TLS

### LINKED ISSUE #956: Pelican fails to start on the first attempt when using TLS certificates with an RSA key
Pelican Version:

```
[root@itb-osdf-pelican-origin-54b4c6c989-bhr8f pelican]# ./osdf --version
Version: dev
Build Date: 2024-03-15T17:55:45Z
Build Commit: d360e742e39c9e9d5dd928cb271464c87c159c68
Built By: goreleaser
```

Pelican Service:
- [ ] Client
- [ ] Plugin
- [X] Registry
- [X] Director
- [X] Origin
- [X] Cache
- [ ] Other (please give the detail)

@williamnswanson and I have definitely seen this issue on the registry and origin and I suspect that this is also a problem for the director and registry.

**Describe the bug**

With an RSA key in `Server.TLSKey`, the `serve` commands will fail:

```
# file tls.key
tls.key: PEM RSA private key
# ./osdf origin serve
Logging.LogLocation is set to /var/log/pelican/pelican.log. All logs are redirected to the log file.
Error: Failure when configuring the server: Private key file, /etc/pelican/cert-orig/tls.key, contains no private key
```

However, subsequent attempts to start the origin appear to succeed. Perhaps related, the TLS CA and key are generated on the initial attempt.

**To Reproduce**

1. Create a TLS cert and RSA key pair and place them in the `Server.TLSCertificate` and `Server.TLSKey` locations
1. Do not create TLS CA files
1. Run `./osdf origin serve`

**Expected behavior**

I expect the Pelican server to start on the first try with an RSA TLS key

### PR DESCRIPTION
**Pull request by haoming29**
Fixes #956 

This PR allows admin to pass RSA key in either PKCS8 or PKCS1 form as the TLSKey or TLSCAKey. Came with tests.

This PR also bypass the error reporting if only TLS cert and TLS key are present and CA cert is missing. It will instead generate a CA with its private key (mostly for xrootd and broker) and continue.

### CODE REVIEW

**Review by turetske — CHANGES_REQUESTED** [review]
Very minor test name change requested

File: config/init_server_creds_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		privateKey, err := LoadPrivateKey(keyLocation, true)
+		require.NoError(t, err)
+		require.NotNil(t, privateKey)
+
+		if _, ok := privateKey.(*rsa.PrivateKey); !ok {
+			assert.Fail(t, "loaded private key type is not *rsa.PrivateKey")
+		}
+	})
+
+	t.Run("rsa-pkcs1-not-allow-error", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (haoming29) [reply config/init_server_creds_test.go:131]: Good call. Fixed in the next commit

File: config/init_server_creds_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		keyLocation := filepath.Join(tempDir, "rsa.key")
+		err := generatePrivateKey(keyLocation, "rsa", "pkcs1")
+		require.NoError(t, err)
+		privateKey, err := LoadPrivateKey(keyLocation, false)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "RSA type private key is not allowed for")
+		require.Nil(t, privateKey)
+	})
+
+	t.Run("rsa-pkcs8-not-allowed", func(t *testing.T) {
```
Comment (turetske) [inline config/init_server_creds_test.go:142]:
```suggestion
	t.Run("rsa-pkcs8-key-not-allowed", func(t *testing.T) {
```

**Review by turetske — CHANGES_REQUESTED** [review]
Just suggest changing a name to be more consistent

**Review by turetske — APPROVED** [review]
LGTM

**Pull request merged**